doc/src/manual/performance-tips.md

```diff
@@ -331,6 +331,14 @@ For reasons of length the results are not shown here, but you may wish to try th
 the type is fully-specified in the first case, the compiler doesn't need to generate any code
 to resolve the type at run-time. This results in shorter and faster code.
 
+One should also keep in mind that not-fully-parameterized types behave like abstract types. For example, even though a fully specified `Array{T,n}` is concrete, `Array` itself with no parameters given is not concrete:
+
+```jldoctest myambig3
+julia> isconcretetype(Array), isabstracttype(Array), isstructtype(Array), isconcretetype(Array{Int}), isconcretetype(Array{Int,1})
+(false, false, true, false, true)
```

Comment:
```suggestion
julia> !isconcretetype(Array), !isabstracttype(Array), isstructtype(Array), !isconcretetype(Array{Int}), isconcretetype(Array{Int,1})
(true, true, true, true, true)
```

Reply:
I'm not sure about this, it seems a bit harder to decode for the reader imho.

Reply:
My think was that it would be helpful that all of these are testing true, so the reader doesn't need to take the negation in their head

Reply:
In this case, to emphasize the all-true-tests, perhaps:
```julia
julia> !isconcretetype(Array) && !isabstracttype(Array) && isstructtype(Array) && !isconcretetype(Array{Int}) && isconcretetype(Array{Int,1})
true
```
?

Reply:
It's just that I find `false` more readable than a `!` in front of a function.